executor/read/tabled_call_executor.rs

```diff
@@ -82,9 +89,10 @@ impl TabledCallExecutor {
         output_batch
     }
 
-    pub(crate) fn create_suspend_point_for(&self, executor_index: ExecutorIndex) -> SuspendPoint {
+    pub(crate) fn create_suspend_point_for(&self, executor_index: ExecutorIndex, depth: usize) -> SuspendPoint {
         SuspendPoint::TabledCall(TabledCallSuspension {
             executor_index,
+            depth,
```

Comment:
Is the `try_read_next_batch` below where we do recursion/cycle detection? By checking the write lock on the executor state? Is is ever possible to hold the lock in non-cyclic cases -> trigger suspend points and recomputation?

Reply:
It shouldn't be possible in a single threaded case.